packages/eslint-config/index.js

```diff
@@ -7,7 +7,8 @@ import tseslint from 'typescript-eslint'
 export default [
   js.configs.recommended,
   nodePlugin.configs['flat/recommended'],
-  ...tseslint.configs.recommended,
+  ...tseslint.configs.strictTypeChecked,
+  ...tseslint.configs.stylisticTypeChecked,
```

Comment:
There's a few [shared configs](https://typescript-eslint.io/users/configs) to choose from.

Their suggestion is to enabling `recommended-type-checked` and `stylistic-type-checked` to start with. Then replacing  `recommended-type-checked` with `strict-type-checked` if you're comfortable working with TypeScript.